src/workspacefactory/wfactory_generator.js

```diff
@@ -135,11 +139,72 @@ WorkspaceFactoryGenerator.prototype.generateWorkspaceXml = function() {
 };
 
 /**
+ * Generates a string representation of XML in JavaScript for toolbox or
+ * pre-loaded workspace blocks. Used for exporting.
+ *
+ * @param {string} xml String representation of XML to be converted to JS.
+ * @param {string} name Name of toolbox or preloaded-workspace.
+ * @param {string} mode Editing mode information. Either
+ *     WorkspaceFactoryController.MODE_TOOLBOX or
+ *     WorkspaceFactoryController.MODE_PRELOAD.
+ * @returns {string} String representation of JavaScript file for export.
+ */
+WorkspaceFactoryGenerator.prototype.generateJsFromXml = function(xml, name, mode) {
+  let jsFromXml = `// If BLOCKLY_${mode}_XML does not exist.
+if (!BLOCKLY_${mode}_XML) {
```

Comment:
Several uses of `BLOCKLY_${mode}_XML`. Define it once. Also, don't forget to escape `name` before outputting it.